peas/sensors.py

```diff
@@ -151,46 +90,35 @@ def capture(self, use_mongo=True, send_message=True):
             sensor_data (dict):     Dictionary of sensors keyed by sensor name.
         """
 
+        # Read from all the readers; we send messages with sensor data immediately, but accumulate
+        # data from all sensors before storing in the db.
+        # Note that there is no guarantee that these are the LATEST reports emitted by the sensors,
+        # as the 
```

Comment:
Comment trails off here. However, regarding the latest, we can clear the input buffer and then it should be the latest, right?

Reply:
Clearing the buffer gets rid of all accumulated input, including the first few bytes of the next line (if it has started emitting them), at which point we'll end up throwing away the "current" reading. I think we can easily solve this problem by splitting the reading of these into two threads, one for each, which disentangles their timing.